lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskAccTest.java

```diff
@@ -0,0 +1,967 @@
+            .build());
+
+    ThrowingCallable call = () -> taskService.createTask(task);
+    ClassificationNotFoundException e =
+        catchThrowableOfType(call, ClassificationNotFoundException.class);
+    assertThat(e.getClassificationKey()).isEqualTo("NOT_EXISTING");
+    assertThat(e.getDomain()).isEqualTo(defaultClassificationSummary.getDomain());
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_ThrowException_When_CreatingTaskWithMissingAttachmentClassificationKey() {
+    Task task = createDefaultTask();
+    ClassificationSummary classificationSummary = new ClassificationSummaryImpl();
+    task.addAttachment(
+        TaskAttachmentBuilder.newAttachment()
+            .classificationSummary(classificationSummary)
+            .objectReference(defaultObjectReference)
+            .channel("E-MAIL")
+            .received(Instant.parse("2018-01-15T00:00:00Z"))
+            .customAttributes(createSimpleCustomPropertyMap(3))
+            .build());
+
+    assertThatThrownBy(() -> taskService.createTask(task))
+        .isInstanceOf(InvalidArgumentException.class)
+        .hasMessageContaining("ClassificationKey of Attachment must not be empty.");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_FetchAttachmentClassification_When_CreatingTaskWithAttachments() throws Exception {
+    Task task = createDefaultTask();
+    ClassificationSummary classificationSummary =
+        ClassificationBuilder.newClassification()
+            .key("T2000")
+            .domain("DOMAIN_A")
+            .type("DOCUMENT")
+            .serviceLevel("P1D")
+            .buildAndStoreAsSummary(classificationService, "admin");
+    ClassificationSummary newClassificationSummary =
+        classificationService.newClassification("T2000", "DOMAIN_A", "").asSummary();
+    task.addAttachment(
+        TaskAttachmentBuilder.newAttachment()
+            .classificationSummary(newClassificationSummary)
+            .objectReference(defaultObjectReference)
+            .channel("E-MAIL")
+            .received(Instant.parse("2018-01-15T00:00:00Z"))
+            .customAttributes(createSimpleCustomPropertyMap(3))
+            .build());
+
+    assertThat(newClassificationSummary.getServiceLevel()).isNull();
+
+    task = taskService.createTask(task);
+    newClassificationSummary = task.getAttachments().get(0).getClassificationSummary();
+
+    assertThat(classificationSummary.getId()).isNotNull();
+    assertThat(classificationSummary.getDomain()).isNotNull();
+    assertThat(classificationSummary.getServiceLevel()).isNotNull();
+  }
+
+  private Map<String, String> createSimpleCustomPropertyMap(int propertiesCount) {
+    return IntStream.rangeClosed(1, propertiesCount)
+        .mapToObj(String::valueOf)
+        .collect(Collectors.toMap("Property_"::concat, "Property Value of Property_"::concat));
+  }
+
+  private Task createDefaultTask() {
+    Task task = taskService.newTask(defaultWorkbasketSummary.getId());
```

Comment:
@gitgoodjhe should we use TaskBuilder in this method instead? One reason to use taskService here is that our tests are intended to show the API. However, I'm not sure if using taskService.newTask(...) is relevant for all create Tests